planning/planning_validator/autoware_planning_validator_collision_checker/src/utils.cpp

```diff
@@ -0,0 +1,237 @@
+  const double time_horizon)
+{
+  autoware_utils::LineString2d trajectory_ls;
+  for (const auto & p : ego_traj.front_traj) {
+    trajectory_ls.emplace_back(p.pose.position.x, p.pose.position.y);
+  }
+
+  auto is_road = [](const lanelet::ConstLanelet & ll) {
+    return ll.hasAttribute(lanelet::AttributeName::Subtype) &&
+           ll.attribute(lanelet::AttributeName::Subtype).value() ==
+             lanelet::AttributeValueString::Road;
+  };
+
+  auto ignore_turning = [&params](const lanelet::ConstLanelet & ll) {
+    if (!ll.hasAttribute("turn_direction")) return false;
+    if (ll.attribute("turn_direction") == "straight") return false;
+    return !params.right_turn.check_turning_lanes;
+  };
+
+  auto extend_lanelet =
+    [&](const lanelet::ConstLanelet & ll, const geometry_msgs::msg::Pose & overlap_point) {
+      const auto distance_th = params.detection_range;
+      lanelet::ConstLanelets extended_lanelets{ll};
+      auto current_arc_length =
+        lanelet::utils::getArcCoordinates(extended_lanelets, overlap_point).length;
+      if (current_arc_length >= distance_th) return extended_lanelets;
+
+      while (current_arc_length < distance_th) {
+        const auto prev_lanelets = route_handler.getPreviousLanelets(ll);
+        if (prev_lanelets.empty()) break;  // No more previous lanelets to extend
+        extended_lanelets.push_back(prev_lanelets.front());
+        current_arc_length += lanelet::utils::getLaneletLength2d(prev_lanelets.front());
+      }
+      std::reverse(extended_lanelets.begin(), extended_lanelets.end());
+      return extended_lanelets;
+    };
+
+  const auto lanelet_map_ptr = route_handler.getLaneletMapPtr();
+  const auto candidates = lanelet_map_ptr->laneletLayer.search(
+    boost::geometry::return_envelope<lanelet::BoundingBox2d>(trajectory_ls));
+  for (const auto & ll : candidates) {
+    const auto id = ll.id();
+    if (
+      !is_road(ll) || ignore_turning(ll) || contains_lanelet(lanelets.trajectory_lanelets, id) ||
+      contains_lanelet(lanelets.connected_lanelets, id))
+      continue;
+
+    const auto overlap_index = get_overlap_index(ll, ego_traj.front_traj, trajectory_ls);
+    if (!overlap_index || overlap_index->first < ego_traj.front_index) continue;
+    const auto mid_idx = (overlap_index->first + overlap_index->second) / 2;
+    const auto overlap_point =
+      lanelet::utils::getClosestCenterPose(ll, ego_traj.front_traj[mid_idx].pose.position);
+    std::pair<double, double> overlap_time;
+    overlap_time.first =
+      rclcpp::Duration(ego_traj.front_traj[overlap_index->first].time_from_start).seconds();
+    overlap_time.second =
+      rclcpp::Duration(ego_traj.back_traj[overlap_index->second].time_from_start).seconds();
+    if (overlap_time.first > time_horizon) continue;
+    lanelets.target_lanelets.emplace_back(
+      ll.id(), extend_lanelet(ll, overlap_point), overlap_point, overlap_time);
+  }
+}
+
+void set_left_turn_target_lanelets(
```

Comment:
I did not try to deeply understand the logic of the functions, but why are the `set_right_turn_target_lanelets` and `set_left_turn_target_lanelets` so different ? I would expect the only change to be the check on the lanelet turn direction.

Reply:
Does this mean we assume right-hand traffic ? I think this should be adjustable with a parameter to allow the use of the module with left-hand traffic.